src/tools.cpp

```diff
@@ -49,6 +56,38 @@ StateHistory sampleKeplerOrbit( const Vector6& initialState,
     return stateHistory;
 }
 
+//! Sample SGP4 orbit
+StateHistory sampleSGP4Orbit( const Tle& tle,
+                              const double propagationTime,
+                              const int numberOfSamples,
+                              const double initialEpochJulian )
+{
+    SGP4 sgp4( tle );
+    Vector6 state;
+    DateTime initialEpoch( ( initialEpochJulian - 1721425.5 ) * TicksPerDay );
```

Comment:
Define 1721425.5 beforehand to minimize the risk of accidentally changing the value.